ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -486,6 +486,11 @@ protected AstNode visitSelect(final Select node, final Void context) {
 
     private void analyzeWhere(final Expression node) {
       analysis.setWhereExpression(node);
+
+      final WhereTypeValidator validator = new WhereTypeValidator(
+          analysis.getFromSourceSchemas(false),
+          metaStore);
+      validator.validateWhereExpression(node);
```

Comment:
Let's move this validation to `LogicalPlanner.buildFilterNode`, as that doesn't need to worry about multiple sources etc. So we're making the problem simpler by moving the validation there.